### ISSUE TITLE
Possibility to disable rules on a per-mode basis

### ISSUE THREAD

**Issue by dschrempf** [emacs-languagetool/lsp-ltex#40]
For example, the `WHITESPACE_RULE` is useful many times, but not in a Git commit buffer.

**Comment by jcs090218** [comment]
Maybe try tweak these variables: 

https://github.com/emacs-languagetool/lsp-ltex/blob/84d9cd89dfb2856d3566ed1930a78dc039b19ab4/lsp-ltex.el#L95-L105

**Comment by dschrempf** [comment]
Thanks. I was assuming these are global variables. I mean we could set them buffer-locally when entering the respective text-based mode.

It would be nice to have this feature available from the LSP action. That is, disable this rule for this major mode.

I tried doing the following:

```
(add-hook 'git-commit-mode-hook
            (lambda () (set (make-local-variable 'lsp-ltex-disabled-rules)
                            '(:en-US ("WHITESPACE_RULE")))))
```

But that doesn't really work, I think because `lsp-ltex--combined-disabled-rules` is not updated accordingly.

**Comment by jcs090218** [comment]
Yeah, you are right. I haven't looked at the code for a long time, so it will take me some time to rework this. 🤔

BTW, PRs are always welcome!
